### PR TITLE
More bug fixes for NXEngine script

### PR DESCRIPTION
**Pull request by PokeEngineer**

### CODE REVIEW

File: scriptmodules/ports/nxengine.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ function configure_nxengine() {
     cat >"$md_inst/NXEngine.sh" << _EOF_
 #!/bin/bash
 cd "$md_inst"
-./nx
+sudo ./nx
```
Comment (zerojay) [inline scriptmodules/ports/nxengine.sh]: Why exactly does this need to run as root?

**Comment by PokeEngineer** [comment]
I can't really give any more detailed explanation than this:
NXEngine extracts data from the Cave Story data files in the directory that it was ran in. If the symlinks in that directory are set to be ran by the root user, running NXEngine as the "pi" user will give you errors.

**Comment by PokeEngineer** [comment]
I just fixed this by making the script set the symlinks to be ran by the user "pi" and ditching running NXEngine as root entirely.

**Comment by zerojay** [comment]
Sounds better to me. Unfortunately, I can't test this myself as my pi is currently dead. I'll merge it for now though.

**Pull request merged**